src/http/http_rpc_context.h

```diff
@@ -339,4 +335,204 @@ namespace enclave
       packed,
```

Comment:
`make_rpc_context` is mostly used as a convenience function for the tests, creating a working context without going through a TLS connection. Maybe we should split this useage into a separate function (always returning a `HttpRpcContext`), separate from the `pbft_config`, `forwarder`, and `node_state` uses, which want to be closer to `HttpRpcEndpoint::handle_request`.

Reply:
Yes, for forwarding I think we must pass a flag telling the remote which it is, because it won't have seen the upgrade handshake. I don't see how things could work otherwise.

Technically, if we stick to not supporting fragmentation and/or de-fragment before we send, we could probably tell them apart because the MSB will always be set on byte 0 of WS frames, whereas none of the first letters of HTTP verbs will have it on...

But just because we can, doesn't mean we should!